drake/systems/plants/RigidBodySystem.cpp

```diff
@@ -185,7 +185,7 @@ class SingleTimeKinematicConstraintWrapper : public Constraint
     rigid_body_constraint->getRobotPointer()->doKinematics(kinsol);
     VectorXd y; MatrixXd dy;
     rigid_body_constraint->eval(nullptr,kinsol,y,dy);
-    initTaylorVecXd(y,dy*autoDiffToGradientMatrix(tq),ty);
+    initializeAutoDiffGivenGradientMatrix(y, (dy * autoDiffToGradientMatrix(tq)).eval(), ty);
```

Comment:
just asking here to make sure we've discussed it once / get a second opinion.  Not requesting any code change.